### DISCUSSION TITLE
Planned Licensing Changes and Updates

### DISCUSSION THREAD

**Discussion by generic-pers0n** [saucedacity/saucedacity#53] [Announcements]
This announcement kind of serves as a reminder that any code you contribute to Saucedacity **must be licensed under the GPLv2 or later**. (I plan to carry keep this policy until the end of time). However, this announcement comes with some changes in roadmap too.

I've had this idea in mind, and now's the time that I reveal it: **We should license Saucedacity/Tenacity under GPLv3 or later at one point**. This is especially given the concern of future license incompatibilities with some libraries, including possibly some Qt libraries. Of course, I know that I won't be able to simply relicense Saucedacity/Tenacity overnight. Therefore, this take some time. As things progress, however, rearchitecting might naturally carry on this process since that's something that we can focus on too.

I will make this clear now: **my contributions to Saucedacity are (and will continue to be) licensed under GPLv2 or later**.

I know that I'm merging code from Tenacity over to Saucedacity, and for now we'll keep the license. Later, we might ask the appropriate contributors to relicense their code under GPLv2 or later. Any existing non-GPLv2 or later code will likely be rewritten for more than just licensing reasons. **An exception is any code coming from Tenacity.**

Finally, **everyone keeps the copyright to their own code, etc. I own the copyright to code I write, not to someone else's hard work.** I'd like to reiterate that due to what happened to upstream

TL;DR: I want to upgrade Saucedacity's license to GPLv3 or later. The natural process of rearchitecting will probably help take care of licensing issues, however.
